### ISSUE TITLE
core dump, how to report it properly?

### ISSUE THREAD

**Issue by habamax** [vim/vim#11368]
### Steps to reproduce

I can't provide steps to reproduce as it happens to a single file with long lines + my config + my regular editing/navigation.

I get core dump message:
```
Vim: Caught deadly signal SEGV
Vim: Finished.
zsh: segmentation fault (core dumped)  vim
```
I understand that this has no actual value so, how should I get core dump and provide it here? (have no experience here, sadly)

### Expected behaviour

No core dump

### Version of Vim

9.0.0747

### Environment

- fedora 36
- zsh
- urxvt

**Comment by chrisbra** [comment]
If this happens often, try running it under a debugger, but make sure to enable a debug build. Or you may try running it with asan enabled.

Check: https://github.com/vim/vim/wiki/Debugging-Vim-Bugs

**Comment by dpelle** [comment]
Please try to build with asan (Address sanitizer) so we get more information.
It only a matter of uncommenting this line in `vim/src/Makefile`:
```
SANITIZER_CFLAGS = -g -O0  -fsanitize-recover=all -fsanitize=address -fsanitize=undefined -fno-omit-frame-pointer
```
Then rebuild:
```
$ cd vim/src
$ make clean; make -j8
```

If a memory error occurs, vim will output stacks on stderr.
You could capture stderr when starting vim with:
```
$ cd vim/src
$ ./vim 2> asan.log
```

**Comment by habamax** [comment]
Thanks @dpelle, here is the log:

[asan.log](https://github.com/vim/vim/files/9787508/asan.log)

**Comment by brammool** [comment]
> Thanks @dpelle, here is the log:
> 
> [asan.log](https://github.com/vim/vim/files/9787508/asan.log)

Thanks, that is a good stack trace.

I guess what happens is that the window has syntax highlighting, but it
never calls syntax_start().  syn_block is then still pointing to another
window, a popup which was closed.

There are quite a few (dangerous) assumptions in this code.  A simple
check would be to pass the window pointer and check it in
syntax_end_parsing().

-- 
The Law of VIM:
For each member b of the possible behaviour space B of program P, there exists
a finite time t before which at least one user u in the total user space U of
program P will request b becomes a member of the allowed behaviour space B'
(B' <= B).
In other words: Sooner or later everyone wants everything as an option.
                                        -- Vince Negri

 /// Bram Moolenaar -- ***@***.*** -- http://www.Moolenaar.net   \\\
///                                                                      \\\
\\\        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by habamax** [comment]
Thank you Bram, looks like it is fixed with `9.0.0750`.
